AndroidProblem/app/src/main/java/com/example/android/androidproblem/MainActivity.kt

```diff
@@ -0,0 +1,65 @@
+package com.example.android.androidproblem
+
+import android.content.SharedPreferences
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Button
+import android.widget.EditText
+import android.widget.ListView
+import android.widget.Toast
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+
+class MainActivity : AppCompatActivity() {
+    private val emailId by lazy { findViewById<EditText>(R.id.email_id) }
+    private val userList by lazy { findViewById<ListView>(R.id.user_list) }
+    private val done by lazy { findViewById<Button>(R.id.done_button) }
+    private val apiService by lazy { RestApiService() }
+    private val gson = Gson()
+    private lateinit var sharedPref: SharedPreferences
+    private val type by lazy { object : TypeToken<ArrayList<User>>() {}.type }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        GlobalScope.launch { getCache() }
+        done.setOnClickListener {
+            if (emailId.text.isNullOrEmpty()) {
+                Toast.makeText(this, R.string.message, Toast.LENGTH_SHORT).show()
+            } else {
+                getHttpResponse()
+            }
+        }
+    }
+
+    private fun getCache() {
+        sharedPref = getSharedPreferences(name, MODE_PRIVATE)
+        sharedPref.let {
+            val userString = it.getString(key, null)
+            if (userString != null) {
+                createListView(gson.fromJson<ArrayList<User>>(userString, type))
+            }
```

Comment:
No I asked you to use `let` in place of `if (userString != null)`, not with sharedPref.
Why do we use `let`?

Reply:
To check nullable types and execute the `let` block only if the value is non-null.